src/firefly/js/templates/fireflyviewer/FireflyViewerManager.js

```diff
@@ -89,6 +89,12 @@ function onAnyAction(layoutInfo, action, views) {
         case CHART_ADD:
         case CHART_REMOVE:
         {
+            if (action.type === TBL_RESULTS_ADDED || action.type ===  TABLE_LOADED) {
+                const tbl = getTblById(action.payload.tbl_id);
+                if (tbl?.request?.META_INFO?.UploadTable) { //tbl?.tableMeta?.UploadTable
+                    return layoutInfo;
+                }
+            }
```

Comment:
This is going to have to be in two or three of the templates, particularly hydra template